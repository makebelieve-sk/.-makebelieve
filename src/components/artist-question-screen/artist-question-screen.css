.game_artist {
    display: flex;
    justify-content: space-between;
    margin-top: 80px; 
}
.artist {
    position: relative;
    width: 134px;
    color: #230d1a;
    text-align: center;
    cursor: pointer; 
}
.artist_picture {
    margin-bottom: 5px;
    width: 134px;
    height: 134px;
    background-color: #f0eed5;
    border: solid 5px #230d1a;
    border-radius: 134px;
    cursor: pointer; 
}
.artist:hover .artist_picture,
.artist_input:focus + .artist_name .artist_picture {
    border-color: #ff9749; 
}
.artist:active .artist_picture {
    border-color: #f00000; 
}