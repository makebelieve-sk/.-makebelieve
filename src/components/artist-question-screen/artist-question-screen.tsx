import React from 'react';

import { questionType, isArtistAnswerType, userAnswerType, isGenreAnswerType } from '../../types';

import './artist-question-screen.css';

type ArtistQuestionScreenType = {
    question: questionType,
    onAnswer: (userAnswerArray: isArtistAnswerType | userAnswerType) => any,
    renderPlayer: (answer: isGenreAnswerType | string, i: number) => any
};

export const ArtistQuestionScreen: React.FC<ArtistQuestionScreenType> = ({ question, onAnswer, renderPlayer }) => {
    const { answers, song } = question;

    return (
        <section className="section-wrapper">
            <section className="game_screen">
                <h2 className="game_title">Кто исполняет эту песню?</h2>
                <div className="game_track">
                    {renderPlayer(song, 0)}
                </div>
                <form className="game_artist">
                    {answers.map((answer: any) => {
                        return (
                            <div className="artist" key={`answer-${answer.id}`}>
                                <input 
                                    className="artist_input visually-hidden" 
                                    type="radio" 
                                    name="answer" 
                                    value={`answer-${answer.id}`} 
                                    id={`answer-${answer.id}`} 
                                    onClick={() => onAnswer(answer)}
                                />
                                <label className="artist_name" htmlFor={`answer-${answer.id}`}>
                                    <img className="artist_picture" src={answer.picture} alt={answer.artist} />
                                    {answer.artist}
                                </label>
                            </div>
                        );
                    })}
                </form>
            </section>
        </section>
    );
};