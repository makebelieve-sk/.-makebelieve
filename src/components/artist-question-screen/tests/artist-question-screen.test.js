import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import { ArtistQuestionScreen } from '../artist-question-screen';

Enzyme.configure({ adapter: new Adapter() });

const setUp = (prop1, prop2, prop3) => shallow(<ArtistQuestionScreen question={prop1} onAnswer={prop2} renderPlayer={prop3} />);

const mock = {
    question: {
        song: {
            artist: `John Snow`,
            src: `https://s3.amazonaws.com/freecodecamp/drums/Heater-1.mp3`
        },
        answers: [
            {
                id: 1,
                picture: `https://placehold.it/134x134`,
                artist: 'John Snow'
            },
            {
                id: 2,
                picture: `https://placehold.it/134x134`,
                artist: 'Jack Daniels'
            },
            {
                id: 3,
                picture: `https://placehold.it/134x134`,
                artist: 'Jim Beam'
            }
        ]
    },
    onAnswer: jest.fn(),
    renderPlayer: jest.fn()
};

describe(`ArtistQuestionScreen component`, () => {
    let component;

    beforeEach(() => {
        component = setUp(mock.question, mock.renderPlayer, mock.onAnswer);
    });

    it(`should contain section wrapper`, () => {
        const wrapper = component.find('.section-wrapper');
        expect(wrapper.length).toBe(1);
    });

    it(`test call renderPlayer function`, () => {
        const renderPlayer = jest.fn();
        const artistQuestionScreen = shallow(
            <ArtistQuestionScreen renderPlayer={renderPlayer} question={mock.question} onAnswer={mock.onAnswer} />
        );

        expect(renderPlayer).toBeCalledTimes(1);
    });

    it(`ArtistQuestionScreen correctly renderer with them prop`, () => {
        const tree = renderer  
            .create(<ArtistQuestionScreen 
                question={mock.question}
                renderPlayer={mock.renderPlayer}
                onAnswer={mock.onAnswer}
            />)
            .toJSON();
    
        expect(tree).toMatchSnapshot();
    });
});