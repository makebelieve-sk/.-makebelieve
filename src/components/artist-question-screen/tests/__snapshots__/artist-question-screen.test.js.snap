// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArtistQuestionScreen component ArtistQuestionScreen correctly renderer with them prop 1`] = `
<section
  className="section-wrapper"
>
  <section
    className="game_screen"
  >
    <h2
      className="game_title"
    >
      Кто исполняет эту песню?
    </h2>
    <div
      className="game_track"
    />
    <form
      className="game_artist"
    >
      <div
        className="artist"
      >
        <input
          className="artist_input visually-hidden"
          id="answer-1"
          name="answer"
          onClick={[Function]}
          type="radio"
          value="answer-1"
        />
        <label
          className="artist_name"
          htmlFor="answer-1"
        >
          <img
            alt="John Snow"
            className="artist_picture"
            src="https://placehold.it/134x134"
          />
          John Snow
        </label>
      </div>
      <div
        className="artist"
      >
        <input
          className="artist_input visually-hidden"
          id="answer-2"
          name="answer"
          onClick={[Function]}
          type="radio"
          value="answer-2"
        />
        <label
          className="artist_name"
          htmlFor="answer-2"
        >
          <img
            alt="Jack Daniels"
            className="artist_picture"
            src="https://placehold.it/134x134"
          />
          Jack Daniels
        </label>
      </div>
      <div
        className="artist"
      >
        <input
          className="artist_input visually-hidden"
          id="answer-3"
          name="answer"
          onClick={[Function]}
          type="radio"
          value="answer-3"
        />
        <label
          className="artist_name"
          htmlFor="answer-3"
        >
          <img
            alt="Jim Beam"
            className="artist_picture"
            src="https://placehold.it/134x134"
          />
          Jim Beam
        </label>
      </div>
    </form>
  </section>
</section>
`;
